### PR TITLE
docs(combineLatest): add details about static version of operator

### PR DESCRIPTION
**Pull request by Podlas29**
**Description:**

Yet another pr with more detailed documentation for operator.

I struggled to meaningfully describe what scheduler actually does with this operator. This is related to my issue here: https://github.com/ReactiveX/rxjs/issues/2350. 

Irony is that I know how this operator behaves with different schedulers, but I have no idea how to describe it without going into implementation details (calling out `ArrayObservable` implementation and the fact that `combineLatest` subscribes to input Observables in its `complete` function). I would really appreciate some help or pointers here.

**Other doc prs**
https://github.com/ReactiveX/rxjs/pull/2322 - here is more background what do I do and why
https://github.com/ReactiveX/rxjs/pull/2326

### CODE REVIEW

**Comment by jayphelps** [comment]
I'm going review this shortly. btw you might reference the reactivex docs as they usually contain excellent operator descriptions since it's mostly a sum of all Rx implementations. http://reactivex.io/documentation/operators/combinelatest.html

Just note that, while rare, RxJS sometimes has differing semantics of certain operators.

File: src/observable/combineLatest.ts [hunk 1]
```diff
@@ -39,14 +39,73 @@ export function combineLatest<R>(...observables: Array<ObservableInput<any> | ((
...
- *
- * @example <caption>Dynamically calculate the Body-Mass Index from an Observable of weight and one for height</caption>
+ * arguments. This is done by subscribing to each Observable in order and,
+ * whenever any Observable emits, collecting an array of the most recent
+ * values from each Observable. So if you pass `n` Observables to operator,
+ * returned Observable will always emit an array of `n` values, in order
+ * corresponding to order of passed Observables (value from the first Observable
+ * on the first place and so on).
+ *
+ * Static version of `combineLatest` accepts either an array of Observables
```
Comment (blesh) [inline src/observable/combineLatest.ts]: Shouldn't the static version of combineLatest be documented along side of that code? This file is for the member method.

Comment (Podlas29) [inline src/observable/combineLatest.ts]: Are you sure? This is `observable` directory, where are all functions that land on `Observable`. Unless we understand `static` differently?

Comment (staltz) [inline src/observable/combineLatest.ts]: @blesh this JSDoc is for the static, as you can see down there we have the tag `@static true` which instructs ESDoc when generating the site.

Comment (blesh) [inline src/observable/combineLatest.ts]: @staltz @Podlas29 right you are. 👍

**Pull request merged**